### DISCUSSION TITLE
Top Pages details fail with error 500

### DISCUSSION THREAD

**Discussion by TitanNano** [plausible/analytics#1476] [Self-Hosted Support] [answered]
We upgraded today from 1.3.0 to 1.4.0 and now the top pages details fail to load most of the time. 

It works with some time ranges, but with most, it doesn't. `Today` works, `7 Days` and `30 Days` does not.

API Call: https://plausible.example.com/api/stats/example.com/pages?period=7d&date=2021-11-19&filters=%7B%7D&limit=100&page=1&detailed=true

Let me know if I should file a bug for this or if there is something we have to adjust in our instance.

**Comment by ukutaht** [comment, marked as answer]
Could you check your logs to try to find the error message?

You can see them with `docker logs <plausible_container_name>`.

**Reply by TitanNano** [reply]
> Server: plausible.example.com:80 (http)
> Request: GET /api/stats/example.com/pages?period=month&date=2021-11-22&filters=%7B%7D&limit=100&page=1&detailed=true
> ** (exit) an exception was raised:
>     ** (ArgumentError) The number of parameters does not correspond to the number of question marks!
>         (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:21: Clickhousex.Codec.Values.encode/3
>         (clickhousex 0.4.0) lib/clickhousex/query.ex:62: DBConnection.Query.Clickhousex.Query.encode/3
>         (db_connection 2.2.2) lib/db_connection.ex:1148: DBConnection.encode/5
>         (db_connection 2.2.2) lib/db_connection.ex:1246: DBConnection.run_prepare_execute/5
>         (db_connection 2.2.2) lib/db_connection.ex:1342: DBConnection.run/6
>         (db_connection 2.2.2) lib/db_connection.ex:539: DBConnection.parsed_prepare_execute/5
>         (db_connection 2.2.2) lib/db_connection.ex:532: DBConnection.prepare_execute/4
>         (clickhouse_ecto 0.2.8) lib/clickhouse_ecto/connection.ex:33: ClickhouseEcto.Connection.prepare_execute/5

**Reply by ukutaht** [reply]
Ah gotcha,

This error was fixed but I haven't cut a patch release yet: https://github.com/plausible/analytics/commit/7364733faf54844281e87271b2ec799e02c4f044

I've added it to my list to release a patched `1.4.1`

**Reply by ukutaht** [reply]
OK, can you try docker tag `plausible/analytics:v1.4.1` and let me know if it fixes the issue?

**Reply by TitanNano** [reply]
yes, it seems to work now.

**Reply by hmzisb** [reply]
Can you please explain how you did it. I am facing the same problem. Server goes down with 500 error and I have to restart everytime.

**Reply by TitanNano** [reply]
@hmzisb did you update to the new `v1.4.1` docker image?

**Reply by hmzisb** [reply]
can you guide me on how to do that? I updated the plausible according to the documentation and still the problem is there. Are you asking me to upgrade docker itself?

**Reply by TitanNano** [reply]
If you use the official [`docker-compose` template](https://github.com/plausible/hosting), you should be able to just pull the images and restart.
